src/lean_dojo/interaction/dojo.py

```diff
@@ -269,21 +269,22 @@ def __enter__(self) -> Tuple["Dojo", State]:
                     raise ex
 
             assert res["error"] is None
+
+            self.start_time = time.monotonic()
+            self._set_timer()
+
             # logger.debug(f"Response: {res}")
             if self.uses_tactics:
                 assert res["tacticState"] != "no goals"
                 init_state = TacticState(
                     self._post_process(res["tacticState"]),
                     res["sid"],
                 )
+                return self, init_state
             else:
                 assert self.uses_commands
-                init_state = CommandState(res["sid"])
-
-            self.start_time = time.monotonic()
-            self._set_timer()
-
-            return self, init_state
+                init_state_cmd = CommandState(int(res["sid"]))
```

Comment:
What is this change for?

Reply:
I was trying to fix this error:
```
Incompatible types in assignment (expression has type "CommandState", variable has type
"TacticState")  [assignment]
```
One can not give one variable two different types. 
But given your upper comment of how the type State is composited, just making them of type State in the first place is the right fix. Thanks for the comment!